From 5d9f8fc2691a2ae9d5105b8b5e711512c875dd5a Mon Sep 17 00:00:00 2001
From: rdonio <ron.d@variscite.com>
Date: Sun, 30 Aug 2015 14:23:48 +0300
Subject: [PATCH] Fix FSL multi patch append bug

---
 .../gstreamer/gstreamer1.0-plugins-bad_%.bbappend                   | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/imx/meta-bsp/recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_%.bbappend b/imx/meta-bsp/recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_%.bbappend
index 446cfef..070f388 100644
--- a/imx/meta-bsp/recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_%.bbappend
+++ b/imx/meta-bsp/recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_%.bbappend
@@ -50,9 +50,9 @@ GPU_PATCHES = " file://1.4.5-Use-viv-direct-texture-to-bind-buffer.patch \
                 file://0008-gl-Fix-glimagesink-loop-playback-failed-in-wayland.patch \
 "
 
-SRC_URI_append_mx6q  = "${GPU_PATCHES}"
-SRC_URI_append_mx6dl = "${GPU_PATCHES}"
-SRC_URI_append_mx6sx = "${GPU_PATCHES}"
+SRC_URI_append_mx6  = "${GPU_PATCHES}"
+#SRC_URI_append_mx6dl = "${GPU_PATCHES}"
+#SRC_URI_append_mx6sx = "${GPU_PATCHES}"
 
 # include fragment shaders
 FILES_${PN}-opengl += "/usr/share/*.fs"
-- 
1.9.1

