From 7822c48234ae7f3161ec6dbc4d4f54008b790e24 Mon Sep 17 00:00:00 2001
From: Eran Matityahu <eran.m@variscite.com>
Date: Mon, 8 Aug 2016 12:07:30 +0300
Subject: [PATCH 1/1] psplash: Change colors for the Variscite Yocto logo

Signed-off-by: Eran Matityahu <eran.m@variscite.com>
---
 psplash-colors.h  |  8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/psplash-colors.h b/psplash-colors.h
index d701089..049d8d9 100644
--- a/psplash-colors.h
+++ b/psplash-colors.h
@@ -20,15 +20,15 @@
 #define _HAVE_PSPLASH_COLORS_H
 
 /* This is the overall background color */
-#define PSPLASH_BACKGROUND_COLOR 0xec,0xec,0xe1
+#define PSPLASH_BACKGROUND_COLOR 0xff,0xff,0xff
 
 /* This is the color of any text output */
-#define PSPLASH_TEXT_COLOR 0x6d,0x6d,0x70
+#define PSPLASH_TEXT_COLOR 0x00,0x56,0xa5
 
 /* This is the color of the progress bar indicator */
-#define PSPLASH_BAR_COLOR 0x6d,0x6d,0x70
+#define PSPLASH_BAR_COLOR 0x00,0x56,0xa5
 
 /* This is the color of the progress bar background */
-#define PSPLASH_BAR_BACKGROUND_COLOR 0xec,0xec,0xe1
+#define PSPLASH_BAR_BACKGROUND_COLOR 0xff,0xff,0xff
 
 #endif
-- 
2.9.2

