diff --git a/plat/freescale/imx8mq/imx8m_bl31_setup.c b/plat/freescale/imx8mq/imx8m_bl31_setup.c
index a91f307..4357be1 100644
--- a/plat/freescale/imx8mq/imx8m_bl31_setup.c
+++ b/plat/freescale/imx8mq/imx8m_bl31_setup.c
@@ -314,8 +314,10 @@ void bl31_platform_setup(void)
 	/* gpc init */
 	imx_gpc_init();
 
+#if 0
 	/* switch DDR frequency to 3200 mts */
 	lpddr4_switch_to_3200();
+#endif
 }
 
 entry_point_info_t *bl31_plat_get_next_image_ep_info(unsigned int type)
