diff --git a/plat/imx/common/imx8m/lpddr4_dvfs.c b/plat/imx/common/imx8m/lpddr4_dvfs.c
index b3ea27b..b19bada 100644
--- a/plat/imx/common/imx8m/lpddr4_dvfs.c
+++ b/plat/imx/common/imx8m/lpddr4_dvfs.c
@@ -20,6 +20,7 @@ void lpddr4_swffc(struct dram_info *info, unsigned int init_fsp,
 	unsigned int mr, emr, emr2, emr3;
 	unsigned int mr11, mr12, mr22, mr14;
 	unsigned int tmp;
+	unsigned int derate_backup[3];
 
 	/* 1. program targetd UMCTL2_REGS_FREQ1/2/3,already done, skip it. */
 
@@ -109,14 +110,17 @@ void lpddr4_swffc(struct dram_info *info, unsigned int init_fsp,
 
 	/* 10. Disable automatic derating: derate_enable */
 	tmp= mmio_read_32(DDRC_DERATEEN(0));
+	derate_backup[0] = tmp;
 	tmp &= ~0x1;
 	mmio_write_32(DDRC_DERATEEN(0), tmp);
 
 	tmp= mmio_read_32(DDRC_FREQ1_DERATEEN(0));
+	derate_backup[1] = tmp;
 	tmp &= ~0x1;
 	mmio_write_32(DDRC_FREQ1_DERATEEN(0), tmp);
 
 	tmp= mmio_read_32(DDRC_FREQ2_DERATEEN(0));
+	derate_backup[2] = tmp;
 	tmp &= ~0x1;
 	mmio_write_32(DDRC_FREQ2_DERATEEN(0), tmp);
 
@@ -346,10 +350,9 @@ void lpddr4_swffc(struct dram_info *info, unsigned int init_fsp,
 		mmio_write_32(DDRC_ZQCTL0(0), tmp);
 	}
 
-	/* 40. re-emable automatic derating: derate_enable */
-	tmp= mmio_read_32(DDRC_DERATEEN(0));
-	tmp &= 0xFFFFFFFE;
-	mmio_write_32(DDRC_DERATEEN(0), tmp);
+	mmio_write_32(DDRC_DERATEEN(0), derate_backup[0]);
+	mmio_write_32(DDRC_FREQ1_DERATEEN(0), derate_backup[1]);
+	mmio_write_32(DDRC_FREQ2_DERATEEN(0), derate_backup[2]);
 
 	/* 41. write 1 to PCTRL.port_en */
 	mmio_write_32(DDRC_PCTRL_0(0), 0x1);
diff --git a/plat/imx/common/imx8m/lpddr4_retention.c b/plat/imx/common/imx8m/lpddr4_retention.c
index 8cc44c8..9cdca3d 100644
--- a/plat/imx/common/imx8m/lpddr4_retention.c
+++ b/plat/imx/common/imx8m/lpddr4_retention.c
@@ -250,7 +250,6 @@ void lpddr4_exit_retention(void)
 		INFO("wait STAT to normal state\n");
 	}
  
-	mmio_write_32(DDRC_DERATEEN(0), 0x00000302);
 
 	mmio_write_32(DDRC_PCTRL_0(0), 0x00000001); 
 	 /* dis_auto-refresh is set to 0 */
