diff --git a/handlers/archive_handler.c b/handlers/archive_handler.c
index ca8aef9..10f0549 100644
--- a/handlers/archive_handler.c
+++ b/handlers/archive_handler.c
@@ -197,7 +197,7 @@ static int install_archive_image(struct img_type *img,
 	TRACE("Installing file %s on %s\n",
 		img->fname, path);
 
-	tf.flags = 0;
+	tf.flags = ARCHIVE_EXTRACT_PERM | ARCHIVE_EXTRACT_OWNER;
 
 	ret = pthread_create(&extract_thread, &attr, extract, &tf);
 	if (ret) {
